### PR TITLE
scsi: translate nvme error to scsi error

### PR DESCRIPTION
**Pull request by uchida2016**
This patch translates NVMe status code to SCSI sense according to NVM Express: SCSI Translation Reference.

http://nvmexpress.org/wp-content/uploads/NVM_Express_-_SCSI_Translation_Reference-1_5_20150624_Gold.pdf

### CODE REVIEW

**Comment by dverkamp-intel** [comment]
Thanks for the patch - this looks like a useful change overall. We will take some time to review it.

One concern I have is the use of NVMe status codes in the bdev layer; however, I don't have a better solution (other than inventing yet another set of status codes specific to bdev), so this is probably acceptable.

File: include/spdk/bdev.h [hunk 1]
```diff
@@ -190,6 +190,7 @@ enum spdk_bdev_io_status {
 	SPDK_BDEV_IO_STATUS_FAILED = -1,
 	SPDK_BDEV_IO_STATUS_PENDING = 0,
 	SPDK_BDEV_IO_STATUS_SUCCESS = 1,
+	SPDK_BDEV_IO_STATUS_NVME_ERROR = 2,
```
Comment (dverkamp-intel) [inline include/spdk/bdev.h]: I'm not sure if we still have any code that relies on it, but the error status should probably be negative (-2 would be fine).

Comment (uchida2016) [inline include/spdk/bdev.h]: I see. I change it to -2. Thanks.

**Comment by uchida2016** [comment]
How is the status of review? Do I need to modify something?

**Comment by benlwalker** [comment]
I think the patch just needs to be rebased onto the latest to resolve a merge conflict.

**Pull request closed without merging**